src/Vector2.elm

```diff
@@ -243,14 +271,15 @@ distanceSquared v w =
 `acos((v dot w)/(|v|*|w|))`
 
     angle (1,0) (2,2) == pi/4    -- or 45°
+
 -}
 angle : Float2 -> Float2 -> Float
 angle v w =
     let
         r =
             dot v w / (length v * length w)
     in
-        if r >= 1 then
-            0
-        else
-            acos r
+    if r >= 1 then
```

Comment:
Is this correct? Wouldn' this give NaN for (0, 0) (0, 0)?
Shouldn't it be
```elm
let
    l = length v * length w
in
    if l == 0 then
        0
    else
        dot v w / l
```
?
If it is correct, then which version is faster?

Reply:
You are correct that that function could use as division by zero test as well.

I checked all the divisions in Vector2 and found that besides angle and normalize, project is also affected.

I would write project as `v dot (normalize w)`.

On second thought it is not clear what angle or normalize should return with zero vectors. At least in angle you can argue that zero is one valid answer for the angle between (0,0) and (0,0). In normalizing the behaviour that I implemented is most convenient for me, but it seems wrong to return a vector with length that is not one.

Reply:
Well mathematically there is no right answer, so I think it makes sense to have a sane default.
But in any case, it should at least be documented.

> I would write project as v dot (normalize w)

This doesn't seem right. You can't get rid of the division.  
I think the best is to also just do a test for zero and in that case return 0.
Which, again, is mathematically nonsense, but it seems like a sane default.